setup_libuv.py

```diff
@@ -1,169 +1,135 @@
-        self.libuv_force_fetch = 0
-        self.libuv_verbose_build = 0
         self.use_system_libuv = 0
 
     def build_extensions(self):
-        self.force = self.force or self.libuv_force_fetch or self.libuv_clean_compile
-
+        sources = SOURCES
+        # XXX Is mingw32 check correct here?!
         if self.compiler.compiler_type == 'mingw32' or self.use_system_libuv:
             self.compiler.add_library('uv')
         else:
-            if sys.platform == 'win32':
-                self.libuv_lib = os.path.join(self.libuv_dir, 'Release', 'lib', 'libuv.lib')
-            else:
-                self.libuv_lib = os.path.join(self.libuv_dir, '.libs', 'libuv.a')
-            self.get_libuv()
-            # Set compiler options
-            self.extensions[0].extra_objects.extend([self.libuv_lib])
             self.compiler.add_include_dir(os.path.join(self.libuv_dir, 'include'))
+            self.compiler.add_include_dir(os.path.join(self.libuv_dir, 'src'))
+            if sys.platform != 'win32':
+                self.compiler.add_include_dir(os.path.join(self.libuv_dir, 'src', 'unix'))
+
+            if sys.platform.startswith('linux'):
+                sources += [
+                    'deps/libuv/src/unix/linux-core.c',
+                    'deps/libuv/src/unix/linux-inotify.c',
+                    'deps/libuv/src/unix/linux-syscalls.c',
+                    'deps/libuv/src/unix/procfs-exepath.c',
+                    'deps/libuv/src/unix/proctitle.c',
+                    'deps/libuv/src/unix/sysinfo-loadavg.c',
+                    'deps/libuv/src/unix/sysinfo-memory.c',
+                ]
+            elif sys.platform == 'darwin':
+                sources += [
+                    'deps/libuv/src/unix/bsd-ifaddrs.c',
+                    'deps/libuv/src/unix/darwin.c',
+                    'deps/libuv/src/unix/darwin-proctitle.c',
+                    'deps/libuv/src/unix/fsevents.c',
+                    'deps/libuv/src/unix/kqueue.c',
+                    'deps/libuv/src/unix/proctitle.c',
+                    'deps/libuv/src/unix/pthread-barrier.c',
+                ]
+
+        self.extensions[0].define_macros.append(('_GNU_SOURCE', 1))
+        if sys.platform != 'win32':
+            self.compiler.define_macro('_LARGEFILE_SOURCE', 1)
+            self.compiler.define_macro('_FILE_OFFSET_BITS', 64)
+
         if sys.platform.startswith('linux'):
+            self.compiler.add_library('dl')
             self.compiler.add_library('rt')
+            self.compiler.add_library('util')
+        elif sys.platform == 'darwin':
+            self.extensions[0].define_macros.append(('_DARWIN_USE_64_BIT_INODE', 1))
+            self.extensions[0].define_macros.append(('_DARWIN_UNLIMITED_SELECT', 1))
+            self.compiler.add_library('util')
         elif sys.platform == 'win32':
             self.extensions[0].define_macros.append(('WIN32', 1))
-            if self.compiler.compiler_type != 'mingw32':
```

Comment:
What exactly? I got a warning that `/LTCG` is not necessary and slows down the linker.

And the warning that LIBCMT is conflicting with other libs is no longer present.